static/src/javascripts/projects/commercial/modules/messenger/type.ts

```diff
@@ -0,0 +1,20 @@
+import { isString } from '@guardian/libs';
+import fastdom from '../../../../lib/fastdom-promise';
+import type { RegisterListener } from '../messenger';
+
+const setType = (type: string, adSlot: Element) =>
+	fastdom.mutate(() => {
+		adSlot.classList.add(`ad-slot--${type}`);
+	});
+
+const init = (register: RegisterListener): void => {
+	register('type', (specs, ret, iframe) => {
+		const adSlot = iframe?.closest('.js-ad-slot');
+
+		if (adSlot && isString(specs)) {
```

Comment:
By introducing this check here we're changing the implementation as well adding types. This should be safe since to add a class to `adSlot` it has to be defined in the first place, and to add a reasonable class name to the slot `specs` should be a string 

Before we tolerated adding `ad-slot--` but I don't think this should be possible.